test/linkify_text_test.dart

```diff
@@ -1 +1,406 @@
+        'Mauritius: +230 555-0678',
+        'Liberia: +231 555-0912',
+        'Sierra Leone: +232 555-0456',
+        'Ghana: +233 555-0789',
+        'Nigeria: +234 555-0123',
+        'Chad: +235 555-0456',
+        'Central African Republic: +236 555-0789',
+        'Cameroon: +237 555-0912',
+        'Cape Verde: +238 555-0345',
+        'São Tomé and Príncipe: +239 555-0567',
+        'Equatorial Guinea: +240 555-0789',
+        'Gabon: +241 555-0123',
+        'Republic of the Congo: +242 555-0456',
+        'Democratic Republic of the Congo: +243 555-0789',
+        'Angola: +244 555-0345',
+        'Guinea-Bissau: +245 555-0567',
+        'British Indian Ocean Territory: +246 555-0789',
+        'Ascension Island: +247 555-0912',
+        'Seychelles: +248 555-0456',
+        'Sudan: +249 555-0123',
+        'Rwanda: +250 555-0345',
+        'Ethiopia: +251 555-0678',
+        'Somalia: +252 555-0789',
+        'Djibouti: +253 555-0456',
+        'Kenya: +254 555-0912',
+        'Tanzania: +255 555-0123',
+        'Zanzibar: +255 24 555-0456',
+        'Uganda: +256 555-0678',
+        'Burundi: +257 555-0912',
+        'Mozambique: +258 555-0345',
+        'Zambia: +260 555-0567',
+        'Madagascar: +261 555-0789',
+        'Réunion: +262 555-0123',
+        'Mayotte: +262 269 555-0345',
+        'Zimbabwe: +263 555-0456',
+        'Namibia: +264 555-0678',
+        'Malawi: +265 555-0789',
+        'Lesotho: +266 555-0123',
+        'Botswana: +267 555-0345',
+        'Eswatini: +268 555-0567',
+        'Comoros: +269 555-0789',
+        'South Africa: +27 555-0912',
+        'Saint Helena: +290 555-0345',
+        'Tristan da Cunha: +290 8 555-0567',
+        'Eritrea: +291 555-0789',
+        'Aruba: +297 555-0456',
+        'Faroe Islands: +298 555-0123',
+        'Greenland: +299 555-0345'
+      ];
+
+      for (final entry in phoneEntries) {
+        final result = LinkifyTextSpans(
+          text: entry,
+          textStyle: customStyle,
+          linkStyle: customPhoneStyle,
+          linkTypes: [LinkType.phone],
+          themeData: themeData,
+        );
+
+        final children = result.children;
+
+        if (children != null) {
+          final splitEntry = entry.split(': ');
+          final country = '${splitEntry[0]}: ';
+          final phoneNumber = splitEntry[1];
+
+          expect((children[0] as TextSpan).text, country);
+          expect((children[1] as TextSpan).text, phoneNumber);
+
+          final textSpan = children[0] as TextSpan;
+          expect(textSpan.style?.color, customStyle.color);
+
+          final linkSpan = children[1] as TextSpan;
+          expect(linkSpan.style?.color, customPhoneStyle.color);
+        }
+      }
+    });
+
+    test('Phone codes for Zones 3–4 countries (Europe)', () {
+      const phoneEntries = [
+        'Greece: +30 210-123-4567',
+        'Netherlands: +31 612-345-6789',
+        'Belgium: +32 2-345-6789',
+        'France: +33 1-234-567-890',
```

Comment:
- [ ] should add more variety cases for: Fr, Ru, 
- [ ] should add test case for Asia zone